config/compression.go

```diff
@@ -0,0 +1,32 @@
+package config
+
+import "strings"
+
+type Compression struct {
+	Request  CompressionInfo `mapstructure:"request"`
+	Response CompressionInfo `mapstructure:"response"`
+}
+
+// CompressionInfo defines what types of compressions are supported
```

Comment:
Nitpick: `compressions` -> `compression algorithms`
Nitpick: Please end this comment with a period. Comments at this level are expected to be documentation and full sentences are appropriate.